app/frontend/src/features/profile/ProfileForm.tsx

```diff
@@ -87,12 +95,32 @@ export default function EditProfileForm() {
       {user ? (
         <>
           <form onSubmit={onSubmit}>
+            <Controller
+              name={"avatarUrl"}
+              control={control}
+              render={({ value }) => {
+                return (
+                  <AvatarInput
+                    className={classes.avatar}
+                    id={"profile-picture"}
+                    name={"image"}
+                    src={value}
+                    onChange={async (values) => {
+                      setValue("avatarUrl", values.thumbnail_url);
+                      setValue("avatarKey", values.key);
```

Comment:
Ok after thinking about it, I think the way to do this is:
- only have `avatarKey` here in the form
- handle the `avatarUrl` and displaying it within the `AvatarInput` with local state
- the [`Controller`](https://react-hook-form.com/api/#Controller) render function also has `onChange(newValue)` as a parameter, so pass that to `AvatarInput`
- then within `AvatarInput` in the "real" `onChange`, you can set the url in the local component state, then call from the props `onChange(avatarKey)`

Let me know if that makes no sense :P

Reply:
It makes sense,
I've fixed it now 👌